packages/nextjs/src/config/wrappers/withSentryServerSideGetInitialProps.ts

```diff
@@ -41,8 +41,9 @@ export function withSentryServerSideGetInitialProps(origGetInitialProps: GetInit
 
       const requestTransaction = getTransactionFromRequest(req!);
       if (requestTransaction) {
+        const dynamicSamplingContext = requestTransaction.getDynamicSamplingContext();
         initialProps._sentryTraceData = requestTransaction.toTraceparent();
-        initialProps._sentryBaggage = serializeBaggage(requestTransaction.getBaggage());
+        initialProps._sentryBaggage = dynamicSamplingContextToSentryBaggageHeader(dynamicSamplingContext);
```

Comment:
This logic is repeated 4x - can we extract into a helper func? Can do this in another PR.